### DISCUSSION TITLE
Template String Types

### DISCUSSION THREAD

**Discussion by mikecann** [statelyai/xstate#1433] [General]
Didnt know where else to put this, but im assuming this is awesome news for typing in XState:

https://github.com/microsoft/TypeScript/pull/40336

Nested states that are accessed via '.' can now be strongly typed :)

**Comment by davidkpiano** [comment]
Absolutely! I think @mattpocock might be exploring this?

**Reply by mattpocock** [reply]
Not yet! Waiting for it to land in stable but it'll be a big win for us when it does.
